src/aggregation/mod.rs

```diff
@@ -239,6 +248,10 @@ pub enum Key {
     F64(f64),
 }
 
+trait MergeFruits {
+    fn merge_fruits(&mut self, other: &Self);
```

Comment:
Can we drop pass other by value? That would remove some clones.

Reply:
yep, done